### ISSUE TITLE
How to train on librispeech

### ISSUE THREAD

**Issue by zhaojingxin123** [NoFish-528/encodec-pytorch#24]
Hello, using LibriTTS960 training about how many epochs, the model can be used. Can you share some config or training information?

**Comment by ZhikangNiu** [comment]
> Hello, using LibriTTS960 training about how many epochs, the model can be used. Can you share some config or training information?

maybe 20 epoch? you can check the hf model.

**Comment by zhaojingxin123** [comment]
Thank you for your answer. 感谢，我训练的非常慢，batchsize=2，log_interval: 100，tensor_cut: 240000，每100batch输出一次log，num_workers: 10，用时1min30秒左右，
然后batchsize=6的时候，用时是4min30s左右，好奇batch改大为什么时间没有变短，怎么能让训练变快呀？再次感谢
Why my training is so slow，

<img width="1002" alt="Snipaste_2024-07-02_14-58-07" src="https://github.com/ZhikangNiu/encodec-pytorch/assets/110162364/4328557e-1e2b-4b19-a541-cd5208d19a27">
My
common:
  save_interval: 2
  test_interval: 5
  log_interval: 100
  max_epoch: 30
  seed: 3401
  amp: False
datasets:
  # 训练测试数据位置
  train_csv_path: '/home/zjx/home/zjx/encodec-pytorch-main/datasets/librispeech_train100h_train.csv'
  test_csv_path: '/home/zjx/home/zjx/encodec-pytorch-main/datasets/librispeech_train100h_test.csv'
  batch_size: 2
  # 320000
  tensor_cut: 240000
  num_workers: 10
  fixed_length: 0
  pin_memory: True
# 是否断点继续训练
checkpoint:
  resume: True
  checkpoint_path: '/home/zjx/home/zjx/encodec-pytorch-main/checkpoints/bs2_cut240000_length0_epoch14_lr0.0003.pt'
  disc_checkpoint_path: '/home/zjx/home/zjx/encodec-pytorch-main/checkpoints/bs2_cut240000_length0_epoch14_disc_lr0.0003.pt'
  # 保存位置
  save_folder: '/home/zjx/home/zjx/encodec-pytorch-main/checkpoints'
  save_location: '${checkpoint.save_folder}/bs${datasets.batch_size}_cut${datasets.tensor_cut}_length${datasets.fixed_length}_' 

optimization:
  lr: 3e-4
  disc_lr: 3e-4

lr_scheduler:
  warmup_epoch: 5

model:
  target_bandwidths: [1.5, 3., 6., 12., 24.]
  sample_rate: 24_000
  channels: 1
  train_discriminator: True
  audio_normalize: True
  filters: 32
  ratios: [8, 5, 4, 2]
  disc_win_lengths: [1024, 2048, 512]
  disc_hop_lengths: [256, 512, 128]
  disc_n_ffts: [1024, 2048, 512]
distributed:
  data_parallel: False
  world_size: 4
  find_unused_parameters: False
  torch_distributed_debug: False
  init_method: tmp

**Comment by zhaojingxin123** [comment]
<img width="835" alt="Snipaste_2024-07-02_15-07-49" src="https://github.com/ZhikangNiu/encodec-pytorch/assets/110162364/21100560-7f98-4368-a05a-26f5a2eb5e8c">

**Comment by ZhikangNiu** [comment]
> Thank you for your answer. 感谢，我训练的非常慢，batchsize=2，log_interval: 100，tensor_cut: 240000，每100batch输出一次log，num_workers: 10，用时1min30秒左右， 然后batchsize=6的时候，用时是4min30s左右，好奇batch改大为什么时间没有变短，怎么能让训练变快呀？再次感谢 Why my training is so slow，
> 
> <img alt="Snipaste_2024-07-02_14-58-07" width="1002" src="https://private-user-images.githubusercontent.com/110162364/344909999-4328557e-1e2b-4b19-a541-cd5208d19a27.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MDQ3MTMsIm5iZiI6MTcxOTkwNDQxMywicGF0aCI6Ii8xMTAxNjIzNjQvMzQ0OTA5OTk5LTQzMjg1NTdlLTFlMmItNGIxOS1hNTQxLWNkNTIwOGQxOWEyNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQwNzEzMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYmExNTBiZmJhNWRiZTRiNWVhYzNhOTkxMzhiMjc1YWE2MmIyOGFjMDJjZThjYzY2ODRjZTY1ZWRmZDdjMDFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MHTpKXzPM4g91_F_ssivNvNMEQv_eUPgjGWBODGz1qE"> My common: save_interval: 2 test_interval: 5 log_interval: 100 max_epoch: 30 seed: 3401 amp: False datasets: # 训练测试数据位置 train_csv_path: '/home/zjx/home/zjx/encodec-pytorch-main/datasets/librispeech_train100h_train.csv' test_csv_path: '/home/zjx/home/zjx/encodec-pytorch-main/datasets/librispeech_train100h_test.csv' batch_size: 2 # 320000 tensor_cut: 240000 num_workers: 10 fixed_length: 0 pin_memory: True # 是否断点继续训练 checkpoint: resume: True checkpoint_path: '/home/zjx/home/zjx/encodec-pytorch-main/checkpoints/bs2_cut240000_length0_epoch14_lr0.0003.pt' disc_checkpoint_path: '/home/zjx/home/zjx/encodec-pytorch-main/checkpoints/bs2_cut240000_length0_epoch14_disc_lr0.0003.pt' # 保存位置 save_folder: '/home/zjx/home/zjx/encodec-pytorch-main/checkpoints' save_location: '
> 𝑐
> ℎ
> 𝑒
> 𝑐
> 𝑘
> 𝑝
> 𝑜
> 𝑖
> 𝑛
> 𝑡
> .
> 𝑠
> 𝑎
> 𝑣
> 𝑒
> 𝑓
> 𝑜
> 𝑙
> 𝑑
> 𝑒
> 𝑟
> /
> 𝑏
> 𝑠
> {datasets.batch_size}_cut
> 𝑑
> 𝑎
> 𝑡
> 𝑎
> 𝑠
> 𝑒
> 𝑡
> 𝑠
> .
> 𝑡
> 𝑒
> 𝑛
> 𝑠
> 𝑜
> 𝑟
> 𝑐
> 𝑢
> 𝑡
> 𝑙
> 𝑒
> 𝑛
> 𝑔
> 𝑡
> ℎ
> {datasets.fixed_length}_'
> optimization: lr: 3e-4 disc_lr: 3e-4
> 
> lr_scheduler: warmup_epoch: 5
> 
> model: target_bandwidths: [1.5, 3., 6., 12., 24.] sample_rate: 24_000 channels: 1 train_discriminator: True audio_normalize: True filters: 32 ratios: [8, 5, 4, 2] disc_win_lengths: [1024, 2048, 512] disc_hop_lengths: [256, 512, 128] disc_n_ffts: [1024, 2048, 512] distributed: data_parallel: False world_size: 4 find_unused_parameters: False torch_distributed_debug: False init_method: tmp

你可以减少tensor_cut到1s ≈ 24000采样点。然后增大batch size训练，另外 numworkers不要设置那么大吧

**Comment by zhaojingxin123** [comment]
感谢大佬指点，这就是试一下

**Comment by zhaojingxin123** [comment]
<img width="646" alt="Snipaste_2024-07-02_16-17-31" src="https://github.com/ZhikangNiu/encodec-pytorch/assets/110162364/12b91549-9fdc-4e13-ac17-460b993a5a69">
![image](https://github.com/ZhikangNiu/encodec-pytorch/assets/110162364/cb03d35a-b8f9-4615-b52b-a2626d6ccd56)
感谢，感谢！！！！！

**Comment by ZhikangNiu** [comment]
> <img alt="Snipaste_2024-07-02_16-17-31" width="646" src="https://private-user-images.githubusercontent.com/110162364/344937597-12b91549-9fdc-4e13-ac17-460b993a5a69.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MTUzMjUsIm5iZiI6MTcxOTkxNTAyNSwicGF0aCI6Ii8xMTAxNjIzNjQvMzQ0OTM3NTk3LTEyYjkxNTQ5LTlmZGMtNGUxMy1hYzE3LTQ2MGI5OTNhNWE2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQxMDEwMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZTFjMmYwMDgzNWVkYThhNTA1NzAxZTU1NWFmNjU5ZWIzYWZiNDZiM2U0N2JkODk1YjcyMWFmNDkxN2I2NzlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.MvVceFrwVXGQ56_RkAD02vj7xXfEARETEjWx7qMw2P0"> ![image](https://github.com/ZhikangNiu/encodec-pytorch/assets/110162364/cb03d35a-b8f9-4615-b52b-a2626d6ccd56) 感谢，感谢！！！！！

但是这样对于音频质量是否有影响我暂时未探索过
